odAnalysis/s-extract-mobility.py

```diff
@@ -0,0 +1,192 @@
+    xID = np.sort(xID)
+    print('..sorting done.' )
+
+    #############################################
+    print('Generating a dictionary of all IDs..' )
+
+    b_dict = {} #initially empty
+
+    iID = xID.astype(int)
+    cnt = 0
+    for x in iID:
+        b_dict[x]=cnt
+        cnt+=1
+    
+    print('..done generating dictionary of length ', cnt,'.' )
+
+    print('Testing the dictionary..' )
+    for x in np.random.choice(iID,size=3):
+        print('..This number ',x, ' is ', (x in b_dict) and ('in') or ('NOT in') , ' the Dictionary!' )
+        print( '..', x, '->', b_dict[x] )
+    
+    print('..test Done.' )
+    
+    
+    #############################################
+    print('Preparing the matrix..' )
+
+    m = len(iID)
+    matrix_od = np.zeros( (m,m) ).astype(int)
+    print(matrix_od)
+
+    print('element (1,2)', matrix_od[1,2] )
+    print('..now we have a ', matrix_od.shape, ' matrix!' )
+    
+    
+    #############################################
+    print('Populating the matrix..' )
+    print('..looping over the sample file..' )
+    cnt = 0
+    for row in s01clean.itertuples():
+        #print('....trying [', int(row.O_ID),',', int(row.D_ID),']','=', row.Value)
+
+        if (row.O_ID in b_dict) and (row.D_ID in b_dict):
+            matrix_od[ b_dict[row.D_ID], b_dict[row.O_ID] ] += row.Value
+            cnt+= 1
+    #    else:
+    #        print('....one entry cannot be added: [', row.O_ID, row.D_ID, ']!' )
+
+    print('..done with {:,}'.format(cnt), ' entries TOTAL!' )
+    print('..here\'s the matrix contents:' )
+    print(matrix_od )
+    print('..root-scaled matrix (power = 1/16):' )
+    print(np.power(matrix_od,0.0625) )
+    
+    
+
+    #############################################
+    print('Saving into different files..' )
+    print('##')
+    foutd = '../out/smart/'
+    fbase = 's-od'
+    fname = foutd+fbase+'-b2m-'+date+'.npy'
+    print('..Saving barangay dictionary bID->mNdx: ',fname  )
+    np.save(fname, np.array( list( b_dict.items() ) ) )
+    print('....to readback:' )
+    print('....bIDtomNdx = dict( list( np.load( FILENAME ) ) )' )    
+    print('##')
+    print('..Saving matrix_od into .npy file..' )
+    fname = foutd+fbase+'-odm-'+date+'.npy'
+    np.save(fname, matrix_od )
+    print('..matrix_od saved to file: ',fname, '..' )
+    print('##')
+    print('..Generating the image..' )
+    fname = foutd+fbase+'-img-'+date+'.png'
+    print('....image will be saved at : ', foutd,'.' )
+    fig=plt.figure()
+    plt.axis('off')
+    p=plt.imshow( np.power(matrix_od,0.0625),interpolation='none',vmin=0,vmax=2,cmap='gnuplot2' )
+    #plt.colorbar(p)
+    plt.imsave(fname, np.power(matrix_od,0.0625),vmin=0,vmax=2,cmap='gnuplot2' )
+    print('..Done. Image output: ', fname)
+    print('..Clearing figure objects..' )
+    plt.close('all') #closing all figures
+    print('..done.')
+    print('DONE.')
+    
+    
+   
+###########possible script for running#######
```

Comment:
Define a simple function call here rather than a comment.

--Just letting you know that we can comment per line of code. You can ignore this comment if you want.

Reply:
Actually scratch paper ko yang dulo na yan. To help me remember how to use the python script file.